### ISSUE TITLE
BalenaEtcher Box86 issue with pi-apps

### ISSUE THREAD

**Issue by Getafix17** [Botspot/pi-apps#90]
After and update Balena Etcher and Box 86 how as corrupted. I tried to uninstall trough GUI but it does not uninstall or install. The GUI window closes and reopens automatically.
Trough terminal uninstall 
```
pi@raspberrypi:~ $ ~/pi-apps/apps/BalenaEtcher/uninstall
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package balena-etcher-electron
failed to purge the deb file!
pi@raspberrypi:~ $ ~/pi-apps/manage uninstall BalenaEtcher
Uninstalling BalenaEtcher
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package balena-etcher-electron
failed to purge the deb file!

Failed to uninstall BalenaEtcher!
```
```
pi@raspberrypi:~ $ ls ~/pi-apps/apps
 Arduino                   Cura                     Retropie
'Back to Chromium v78'     Discord                  Scrcpy
 BalenaEtcher              eDEX-UI                 'Sonic Pi 3.2'
 Box86                    'Email Checker'           SpeedTest-CLI
'Boxy SVG'                 FreeCAD                  TBOPlayer
'Chromium Media Edition'  'FreeCAD (precompiled)'   template
'Chromium Widevine'        LibrePCB                'Tor Browser'
'Clam Antivirus'          'Minecraft Bedrock'      'Update Buddy'
'Color Emoji font'        'Minecraft Java'          USBImager
 CommanderPi               piKiss                  'Visual Studio Code'
 Conky                    'Pi Power Tools'         'Windows 10 Theme'
'Cool Retro Term'          PrusaSlicer              Zoom
 Cordless                  Raspi2png
pi@raspberrypi:~ $ ~/pi-apps/uninstall
Removing menu button...

```

**Comment by Getafix17** [comment]
pi@raspberrypi:~ $ ~/pi-apps/manage update-all
Cloning into 'pi-apps'...
remote: Enumerating objects: 77, done.
remote: Counting objects: 100% (77/77), done.
remote: Compressing objects: 100% (63/63), done.
remote: Total 1797 (delta 28), reused 46 (delta 13), pack-reused 1720
Receiving objects: 100% (1797/1797), 11.06 MiB | 1.20 MiB/s, done.
Resolving deltas: 100% (1024/1024), done.
c2612edebfc4ade038ea4c78c5318d24fc371b90
c2612edebfc4ade038ea4c78c5318d24fc371b90
Arduino is identical to the online version. Nothing to do!
f106e3b4d50a8bc04883422f69b4527935c7f0d0
f106e3b4d50a8bc04883422f69b4527935c7f0d0
Back to Chromium v78 is identical to the online version. Nothing to do!
d03cf2a45bf1d9fb2cd7865199cdfe9cad841150
d03cf2a45bf1d9fb2cd7865199cdfe9cad841150
BalenaEtcher is identical to the online version. Nothing to do!
84ac6f710a6c534e038ff899a34d5aa731bcbcd6
84ac6f710a6c534e038ff899a34d5aa731bcbcd6
Box86 is identical to the online version. Nothing to do!
fe198dc0714e51c1048c72ea2c1395281d565644
fe198dc0714e51c1048c72ea2c1395281d565644
Boxy SVG is identical to the online version. Nothing to do!
457859edc8503ada1adf0c571d89f8ca9dd8f03f
457859edc8503ada1adf0c571d89f8ca9dd8f03f
Chromium Media Edition is identical to the online version. Nothing to do!
5d7272c37ac6b1ba63564d00d4c8a8b4710111fd
5d7272c37ac6b1ba63564d00d4c8a8b4710111fd
Chromium Widevine is identical to the online version. Nothing to do!
87118f5266a861d089421252decd796000b9d563
87118f5266a861d089421252decd796000b9d563
Clam Antivirus is identical to the online version. Nothing to do!
d8c0ba512cef73516d9567e59f175d563f9a1d60
d8c0ba512cef73516d9567e59f175d563f9a1d60
Color Emoji font is identical to the online version. Nothing to do!
47bd4b1887e88d71a470a62af3870915a1fe037d
47bd4b1887e88d71a470a62af3870915a1fe037d
CommanderPi is identical to the online version. Nothing to do!
9b44c1ce51144e1a3db9de07efb8fb10a98a9a1e
9b44c1ce51144e1a3db9de07efb8fb10a98a9a1e
Conky is identical to the online version. Nothing to do!
ea066004528f0904f9cd8862b33f82eaaa412199
ea066004528f0904f9cd8862b33f82eaaa412199
Cool Retro Term is identical to the online version. Nothing to do!
f54d2987ed4d4af6f4ed175dd854eed8fa4b1152
f54d2987ed4d4af6f4ed175dd854eed8fa4b1152
Cordless is identical to the online version. Nothing to do!
3e9359ca0119eacedb91c54c28533a4302051d7e
3e9359ca0119eacedb91c54c28533a4302051d7e
Cura is identical to the online version. Nothing to do!
a4b5dee406257a12744edee583b245c4f32d0f08
a4b5dee406257a12744edee583b245c4f32d0f08
Discord is identical to the online version. Nothing to do!
875c329e22a4e67bce1a95a2cba66d855dacaffa
875c329e22a4e67bce1a95a2cba66d855dacaffa
eDEX-UI is identical to the online version. Nothing to do!
80d0d3a20b017fd734147830bf5653cf145005f1
80d0d3a20b017fd734147830bf5653cf145005f1
Email Checker is identical to the online version. Nothing to do!
1d2660c79bc876b9e45c804366040659a9f097b7
1d2660c79bc876b9e45c804366040659a9f097b7
FreeCAD is identical to the online version. Nothing to do!
84c04b24d2c9d54d3046c5d04d004c9419446022
84c04b24d2c9d54d3046c5d04d004c9419446022
FreeCAD (precompiled) is identical to the online version. Nothing to do!
20bf48b124921cc5d7a20a546995044b041f887b
20bf48b124921cc5d7a20a546995044b041f887b
LibrePCB is identical to the online version. Nothing to do!
097266db3296822bb8f316ed3905656624cbbc42
097266db3296822bb8f316ed3905656624cbbc42
Minecraft Bedrock is identical to the online version. Nothing to do!
f1f3931e2d0cb312fee71d4e03243986a022111e
f1f3931e2d0cb312fee71d4e03243986a022111e
Minecraft Java is identical to the online version. Nothing to do!
e7fcc40155e7474814c8d377fbec2536faa41c6d
e7fcc40155e7474814c8d377fbec2536faa41c6d
piKiss is identical to the online version. Nothing to do!
7d8e2da31ea58518522286d3c268367163ba33c7
7d8e2da31ea58518522286d3c268367163ba33c7
Pi Power Tools is identical to the online version. Nothing to do!
299d72bcdb34c0012c1063c50adfbf2981017a4c
299d72bcdb34c0012c1063c50adfbf2981017a4c
PrusaSlicer is identical to the online version. Nothing to do!
f4503582ce8e20d86c8bfa3ef7df117028ac6a51
f4503582ce8e20d86c8bfa3ef7df117028ac6a51
Raspi2png is identical to the online version. Nothing to do!
ee2c1a4c3d1362e68084ff8be474f953a30a707c
ee2c1a4c3d1362e68084ff8be474f953a30a707c
Retropie is identical to the online version. Nothing to do!
d064cab6b7148ab900ea36329a36beb3c4cbfbd3
d064cab6b7148ab900ea36329a36beb3c4cbfbd3
Scrcpy is identical to the online version. Nothing to do!
a6d836df53d2e4b71a79e909d6d9e8711914338e
a6d836df53d2e4b71a79e909d6d9e8711914338e
Sonic Pi 3.2 is identical to the online version. Nothing to do!
533a6c3b36e780fd782d5d099ee6bbaf1ce3ee67
533a6c3b36e780fd782d5d099ee6bbaf1ce3ee67
SpeedTest-CLI is identical to the online version. Nothing to do!
423799048eb2a186e534539dc92428c21794b5c4
423799048eb2a186e534539dc92428c21794b5c4
TBOPlayer is identical to the online version. Nothing to do!
62f4c35685746b36da15e8c3c492cbc43ffb68e7
62f4c35685746b36da15e8c3c492cbc43ffb68e7
template is identical to the online version. Nothing to do!
98164fede54372c92b7b8ffd95a15aa77b69c760
98164fede54372c92b7b8ffd95a15aa77b69c760
Tor Browser is identical to the online version. Nothing to do!
9406c242aa443b85efdbfe4ae177c50b547aef7c
9406c242aa443b85efdbfe4ae177c50b547aef7c
Update Buddy is identical to the online version. Nothing to do!
610c1649fd0a4aa64fce622ba3a0084d612829bf
610c1649fd0a4aa64fce622ba3a0084d612829bf
USBImager is identical to the online version. Nothing to do!
25f37633e1d5b0b9e785dcc3034f68e145481b1c
25f37633e1d5b0b9e785dcc3034f68e145481b1c
Visual Studio Code is identical to the online version. Nothing to do!
71c10c76e6660d0a6f40e1b4dc1371c8e0c9d188
71c10c76e6660d0a6f40e1b4dc1371c8e0c9d188
Windows 10 Theme is identical to the online version. Nothing to do!
00a57209f7a0843dd3b4cfa8b7b292eb9f20989b
00a57209f7a0843dd3b4cfa8b7b292eb9f20989b
Zoom is identical to the online version. Nothing to do!
Updatable: .EOU
updating .
/home/pi/pi-apps/manage update .
Cloning into 'pi-apps'...
remote: Enumerating objects: 77, done.
remote: Counting objects: 100% (77/77), done.
remote: Compressing objects: 100% (63/63), done.
remote: Total 1797 (delta 28), reused 46 (delta 13), pack-reused 1720
Receiving objects: 100% (1797/1797), 11.06 MiB | 1.75 MiB/s, done.
Resolving deltas: 100% (1024/1024), done.
e785a4df1851e14ba831298ad177945b8ea8a902
e785a4df1851e14ba831298ad177945b8ea8a902
No install script found for the . app! Please report this to Botspot.
pi@raspberrypi:~ $

**Comment by Getafix17** [comment]
I am running Raspberry Pi OS 32 bit from an SD card and a 240 GB SSD bot have the same problem after updating Pi-apps. Could you please suggest how to uninstall the apps or maybe do a fresh install.

**Comment by Botspot** [comment]
Sure thing. Thanks for the debug info.  
First, I noticed you "uninstalled" pi-apps. Please install it again to re-create the menu buttons.  
Second, it appears balenaEtcher did uninstall correctly, but somehow failed later on. Without that log, I'm not sure exactly what went wrong, but please **try installing balenaEtcher**.  

For box86, could you first try to uninstall it? Then try to install it.

**Comment by Botspot** [comment]
@Getafix17 Any updates?

**Comment by Getafix17** [comment]
@Botspot.
I reinstalled Pi Apps like you said. I am still having the same issue.
I cannot install any new apps if I hit install from the GUI. The window just closes and reopens. I tried using manage install but that didn't work either. Error message says(All is Well) Folder not found . This happens with all the apps In Pi-apps.  Update all apps via terminal, shows the apps in Pi-apps folder but nothing installs via GUI or terminal. 
This issue I noticed is only with Raspberry pi OS that I have on an SSD and also on two SD cards. Twister OS however I have no issues, I have it also on a SSD and SD card both work fine.

**Comment by Botspot** [comment]
This may be related to #52 .  
Let's check which terminal your OS wants to launch if it's given a choice.  

Please run `readlink -f /usr/bin/x-terminal-emulator`

**Comment by Getafix17** [comment]
pi@raspberrypi:~ $ readlink -f /usr/bin/x-terminal-emulator
/usr/bin/lxterm

**Comment by Botspot** [comment]
lxterm... That's a new one.

I'll add a fix soon.

**Comment by Getafix17** [comment]
Okay thanks!

**Comment by Botspot** [comment]
Wait - did you mean to say `/usr/bin/lxterminal`? Or `/usr/bin/xterm`? 

I've never heard of lxterm.

**Comment by Getafix17** [comment]
I ran the command twice. 
pi@raspberrypi:~ $ readlink -f /usr/bin/x-terminal-emulator
/usr/bin/lxterm
pi@raspberrypi:~ $ readlink -f /usr/bin/x-terminal-emulator
/usr/bin/lxterm
pi@raspberrypi:~ $

**Comment by Botspot** [comment]
That is very weird. Research time!

**Comment by Botspot** [comment]
Oh I see. It's another name for `/usr/bin/xterm`.

**Comment by Botspot** [comment]
Okay, I think I've fixed this issue as of commit a28ec27c733c536a72506feb7bfde62eff48551f.

**Comment by Getafix17** [comment]
Fantastic! it works beautifully after the update. Thanks!
